crates/autopilot/src/solvable_orders.rs

```diff
@@ -261,7 +261,16 @@ impl SolvableOrdersCache {
         counter.record(&auction.orders);
 
         if self.store_in_db {
-            let _id = self.database.replace_current_auction(&auction).await?;
+            let auction_id = self.database.replace_current_auction(&auction).await?;
+            self.database
+                .insert_auction_prices(auction_id, &auction.prices)
```

Comment:
But both pieces of code use ` database::auction_prices::insert` .  Does that mean we're storing in the same table first the external prices and then the settlement prices? That's confusing. I'd probably use separate tables to make the distinction clear.

Reply:
It's external prices in both cases. We don't save settlement prices (if by settlement prices you mean clearing prices)

Reply:
What is the reason then that we insert prices again when they were already inserted in solvable_orders.rs ? If it is the same prices?

Reply:
In `solvable_orders.rs` we save external_prices for all orders in auction.
In `on_settlement_event_updater` we want to eliminate all external prices not related to the orders not contained in the mined transaction.

So, instead of deleting and inserting reduced list, I can just delete (full_list - reduced_list) if you think that is better. with current solution I just wanted to avoid defining new delete function that would take a list of <auction_id, token>

Reply:
I think it's fine to implement it the way you have done but the intention is not clear. This should get a comment somewhere.

Reply:
Added comment in two places.